### PR TITLE
fix(runtime): restore SIGSEGV signal handler for non-wasm platforms

### LINKED ISSUE #1357: `go/types` scope.Insert got `segmentation fault`
```go
package main

import (
	"fmt"
	"go/types"
)

func main() {
	fmt.Println("=== Scope Insert Test ===\n")

	// Create a scope
	scope := types.NewScope(nil, 0, 0, "test")

	// Test 1: Successful insert
	obj1 := types.NewVar(0, nil, "x", types.Typ[types.Int])
	if alt := scope.Insert(obj1); alt != nil {
		panic(fmt.Sprintf("unexpected: insert failed with conflict: %v", alt))
	}
	fmt.Printf("✓ Insert successful: %s\n", obj1.Name())

	// Test 2: Duplicate insert (will fail)
	obj2 := types.NewVar(0, nil, "x", types.Typ[types.String])
	if alt := scope.Insert(obj2); alt != nil {
		fmt.Printf("✓ Insert failed as expected: %s already exists\n", alt.Name())
	} else {
		panic("unexpected: duplicate insert should have failed")
	}

	// Test 3: Insert different name
	obj3 := types.NewVar(0, nil, "y", types.Typ[types.String])
	if alt := scope.Insert(obj3); alt != nil {
		panic(fmt.Sprintf("unexpected: insert failed with conflict: %v", alt))
	}
	fmt.Printf("✓ Insert successful: %s\n", obj3.Name())

	// View results
	fmt.Printf("\nScope contains %d objects\n", scope.Len())

	// Verify expected count
	expectedCount := 2 // x and y
	if scope.Len() != expectedCount {
		panic(fmt.Sprintf("unexpected: scope should contain %d objects, but has %d", expectedCount, scope.Len()))
	}
	fmt.Printf("✓ Scope length check passed\n")

	fmt.Println("\nAll objects:")
	for _, name := range scope.Names() {
		obj := scope.Lookup(name)
		fmt.Printf("  - %s: %v\n", name, obj.Type())
	}
}
```

expect got
```bash
=== Scope Insert Test ===

✓ Insert successful: x
✓ Insert failed as expected: x already exists
✓ Insert successful: y

Scope contains 2 objects
✓ Scope length check passed

All objects:
  - x: int
  - y: string
```
but got follow with `llgo run`
```bash
=== Scope Insert Test ===

signal: segmentation fault
```

### PR DESCRIPTION
**Pull request by luoliwoshang**
## Summary

This PR fixes Issue [#1364](https://github.com/goplus/llgo/issues/1364) by implementing a SIGSEGV signal handler that converts segmentation faults from nil pointer dereferences into recoverable Go panics, matching standard Go runtime behavior.

## Changes

1. **Add SIGSEGV signal handler** (`runtime/internal/runtime/z_rt_default.go`)
   - Converts SIGSEGV signals to Go panics with standard error message
   - Enables recover() to catch nil pointer dereference errors
   - Uses build tag `!wasm && !baremetal` to exclude wasm and baremetal platforms

2. **Add regression test** (`_demo/go/recover-nil/main.go`)
   - Verifies nil pointer dereference can be recovered
   - Ensures signal handler works correctly on non-wasm platforms

### Why use hardcoded constant `SIGSEGV = 0xb`?

SIGSEGV has the same value (11 / 0xb) across all Unix-like systems (Linux, Darwin, BSD, Solaris, etc.). To avoid unnecessary dependencies, we define the constant directly instead of importing from syscall packages.

**Dependency issues when importing syscall:**

Using `import "syscall"` causes linking failures in c-shared/c-archive build modes:

```bash
ld64.lld: error: undefined symbol: internal/reflectlite.Value.IsNil
ld64.lld: error: undefined symbol: internal/reflectlite.TypeOf
ld64.lld: error: undefined symbol: internal/reflectlite.init
ld64.lld: error: undefined symbol: syscall.init
```

Using `import "github.com/goplus/llgo/runtime/internal/clite/syscall"` also fails:

```bash
ld64.lld: error: undefined symbol: internal/reflectlite.TypeOf
ld64.lld: error: undefined symbol: internal/reflectlite.init
ld64.lld: error: undefined symbol: sync.init
```

**Root cause**: Both syscall packages import the `errors` package, which depends on `internal/reflectlite` and `sync`. In c-shared/c-archive build modes, these internal packages are not fully linked by llgo's linker, causing undefined symbol errors.

**Solution**: Use hardcoded constant `SIGSEGV = 0xb` to completely avoid the dependency chain.

## Testing

Tested on all build modes:
- ✅ exe mode: `llgo run` - works
- ✅ c-shared mode: `llgo build -buildmode=c-shared` - works
- ✅ c-archive mode: `llgo build -buildmode=c-archive` - works

## Related

- Fixes #1357
- See PR #1059 for wasm platform requirements

### CODE REVIEW

File: runtime/internal/runtime/z_signal.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//go:build !wasm
+
+package runtime
```
[the remark this answers is not in the document]
Comment (luoliwoshang) [reply runtime/internal/runtime/z_signal.go:19]: done

File: runtime/internal/runtime/z_rt_default.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//go:build !wasm
```
[the remark this answers is not in the document]
Comment (luoliwoshang) [reply runtime/internal/runtime/z_rt_default.go]: done

**Comment by luoliwoshang** [comment]
@xushiwei

**Review by visualfc — APPROVED** [review]
